Multiplatform/Navigation/Sections/AlbumsView.swift

```diff
@@ -48,16 +50,20 @@ struct AlbumsView: View {
             SortSelector(ascending: $sortAscending, sortOrder: $sortOrder)
         }
         .task {
-            await fetchAlbums()
+            if albums.isEmpty {
+                await fetchAlbums()
+            }
         }
         .refreshable {
             reset()
             await fetchAlbums()
         }
         .onChange(of: sortState) {
-            Task {
-                reset()
-                await fetchAlbums()
+            searchTask?.cancel()
+            searchTask = Task {
+                try await Task.sleep(nanoseconds: UInt64(0.5 * TimeInterval(NSEC_PER_SEC)))
+                await fetchAlbums(shouldReset: true)
```

Comment:
Why not put the `reset` method call before this and avoid the additional method `loadMore`?

Reply:
Explained in this commit message:

https://github.com/rasmuslos/AmpFin/pull/34/commits/d906768b47aa9c58a2cf9e15ea74f556840ac561

There will be concurrency issue and the reset has to be done in the fetchAlbums method. Otherwise you will have duplicated items. If you first open the albums view in offline tab, then goto online tab, then change the sort order, then switch back to offline tab, you will see your albums duplicated, this is because when you are doing this this onChange handler is going to be called multiple times and each time the reset is called before the fetchAlbums call, then fetchAlbums call A will first insert its entries, fetchAlbums call B will insert its entries after.

Reply:
The timing would look like this:

Tab switched-> onChangeA and onChangeB called at the same time -> resetA and resetB completed -> fetchA and fetchB started, and pushing their entries -> the final list contains duplicated entires.

Reply:
Then I think it is better to remove the `reset` method and just include the code in the `fetchAlbums` method, as they are only called together.

Also these are purely stylistic, but it may be better to not provide a default value for the `shouldReset` parameter to make calls more explicit and to use a anonymous method for the callback to avoid the additional method that is only used once.